bdk_rpc_example/src/main.rs

```diff
@@ -0,0 +1,182 @@
+mod rpc;
+
+use std::{collections::BTreeMap, path::PathBuf, sync::mpsc::channel};
+
+use bdk_cli::{
+    anyhow,
+    clap::{self, Args, Subcommand},
+    log,
+};
+use bdk_core::{chain_graph::ChainGraph, BlockId, TxHeight};
+use bdk_keychain::{KeychainScan, KeychainTxOutIndex};
+use bitcoincore_rpc::Auth;
+use rpc::{Client, RpcData};
+
+#[derive(Args, Debug, Clone)]
+struct RpcArgs {
+    /// RPC URL
+    #[clap(env = "RPC_URL", long, default_value = "127.0.0.1:8332")]
+    url: String,
+    /// RPC auth cookie file
+    #[clap(env = "RPC_COOKIE", long)]
+    rpc_cookie: Option<PathBuf>,
+    /// RPC auth username
+    #[clap(env = "RPC_USER", long)]
+    rpc_user: Option<String>,
+    /// RPC auth password
+    #[clap(env = "RPC_PASS", long)]
+    rpc_password: Option<String>,
+}
+
+impl From<RpcArgs> for Auth {
+    fn from(args: RpcArgs) -> Self {
+        match (args.rpc_cookie, args.rpc_user, args.rpc_password) {
+            (None, None, None) => Self::None,
+            (Some(path), _, _) => Self::CookieFile(path),
+            (_, Some(user), Some(pass)) => Self::UserPass(user, pass),
+            (_, Some(_), None) => panic!("rpc auth: missing rpc_pass"),
+            (_, None, Some(_)) => panic!("rpc auth: missing rpc_user"),
+        }
+    }
+}
+
+#[derive(Subcommand, Debug, Clone)]
+enum RpcCommands {
+    /// Scans blocks from bitcoin-core RPC and records relevant transactions
+    Scan {
+        /// Block height in which to start intial scan
+        #[clap(long, default_value = "0")]
+        initial_height: u64,
+        /// The unused-script gap will be kept at this value
+        #[clap(long, default_value = "5")]
+        stop_gap: usize,
+    },
```

Comment:
What bout `Sync` which uses the last checkpoint as initial height and doesn't have a stop gap (just scans already derived scripts).

Reply:
I currently don't think RPC requires a difference between sync/scan. Either way we still need to look through every block, starting from the "point of agreement". Stop gap can be set to 0 if we are good to work with current cached scripts.

Let me know if I am missing something.

Reply:
Ahh right. Then we shouldn't call this "stop_gap" since we don't stop if we find a gap right? "address_look_ahead" or something.